src/main/java/jenkins/plugins/slack/SlackNotifier.java

```diff
@@ -390,7 +410,7 @@ public SlackNotifier(final String baseUrl, final String teamDomain, final String
     }
 
     public SlackNotifier(final String baseUrl, final String teamDomain, final String authToken, final boolean botUser, final String room, final String tokenCredentialId,
-                         final String sendAs, final boolean startNotification, final boolean notifyAborted, final boolean notifyFailure,
+                         final String sendAs, final String iconEmoji, final String username, final boolean startNotification, final boolean notifyAborted, final boolean notifyFailure,
```

Comment:
This is unwieldy, and you would need "another" deprecated SlackNotifier to avoid binary incompatibility.
Perhaps we ought to switch to a builder pattern?
Or is the only used for testing? Perhaps mark it as `NoExternalAccess`
@timja WDYT?

Reply:
I think we need a Pojo with a builder pattern, we can't keep adding arguments here.

Reply:
Help me understand a little bit: I get the builder pattern bit, but I don't know where it will actually be used.  Wouldn't I need to make another constructor anyway to avoid the binary incompatibility?

Reply:
There will be one new constructor, and a conversion in the old one to the builder.

Never more will there be a new constructor, there's about 6 in this file